src/Compilers/Core/MSBuildTask/Csc.cs

```diff
@@ -724,7 +726,7 @@ protected override bool CallHostObjectToExecute()
 
             ICscHostObject cscHostObject = HostObject as ICscHostObject;
```

Comment:
:bulb: An `InvalidCastException` is probably clearer than a `NullReferenceException`. I believe the following resolves this without requiring the non-null assertion operator.

```
ICscHostObject cscHostObject = (ICscHostObject)HostObject;
```